docs/metrics/bleu.md

```diff
@@ -0,0 +1,132 @@
+    | 4 | 3 / 14 = 0.214 |
+
+    </center>
+
+    So, our n-Gram overlap is:
+
+    $$
+    0.824^{0.25}\cdot0.625^{0.25}\cdot0.400^{0.25}\cdot0.214^{0.25} = 0.458
+    $$
+
+??? example "Step 3: Calculate Brevity Penalty"
+    We apply a brevity penalty to prevent the BLEU score from giving undeservingly high scores for short generated texts. Recall that our formula is:
+
+    $$
+    \min\left(1, \exp\left(1 - \frac{\text{reference length}}{\text{output length}}\right)\right)
+    = \min\left(1, \exp\left(1 - \frac{\text{23}}{\text{17}}\right)\right)
+    = 0.703
+    $$
+
+??? example "Step 4: Calculate BLEU"
+    Combining our n-Gram overlap and Brevity Penalty, our final BLEU score is:
+
+    $$
+    \text{BLEU} = \text{Brevity Penalty} \times \text{n-Gram Overlap} = 0.458 \times 0.703 = 0.322
+    $$
+
+    Note that in most cases, we may take the average of the BLEU score with respect to multiple reference texts - since multiple interpretations of the same sentences can be allowed. For example, if we calculated the BLEU score with the reference text being "Kolena is a comprehensive machine learning testing and debugging platform to surface hidden model behaviors and demystify model development", our BLEU score would be 0.571 - a much higher score from a small semantic change.
+
+
+## Advantages and Limitations
+Though BLEU is a popular metric in NLP workflows, it comes with its limitations.
+
+1. BLEU fails to consider the semantics of texts. As seen in the example, simply changing "take the mystery out of" to "demistify" — while the text still retains the exact same meaning — yields a much better score, going from 0.3 to 0.6. In contrast with embeddings-based metrics like BertScore, n-gram-based metrics like BLEU only consider the words in the candidate text, rather than the meaning. However, this is addressed by providing multiple possible reference texts when calculating the BLEU score.
+2. BLEU does not consider the order of words. Across a large candidate/reference, BLEU is unable to consider to order and fluency of the sentences due to its short context window of 4-grams. Although BLEU can be extended to include larger n-gram clipped precisions, it would negatively affect shorter texts. Similarly, BLEU does not consider the importance of words in a sentence either. It weighs unimportant words like "the", "an", "too", as much as it would the nouns and verbs in the sentence. Once again, these pitfalls are addressed by embeddings-based metrics like BertScore.
+
+That being said, the metric still has its advantages. It is quick and easy to compute, as opposed to other metrics like BertScore which would take significantly longer to compute and is not easy to justify. Furthermore, it is relatively similar to human judgement (as seen in the [figure below](https://aclanthology.org/P02-1040.pdf)), and is commonly used within NLP which allows you to easily benchmark your models with others and identify pain points.
```

Comment:
Would you say BLEU is more similar to human judgement than other metrics like BertScore or ROUGE?

Reply:
In terms of similarity, it's BertScore > METEOR > BLEU/ROUGE. I'll add that to the docs!